can-define.js

```diff
@@ -239,9 +239,9 @@ define.property = function(objPrototype, prop, definition, dataInitializers, com
 	if ((definition.value !== undefined || definition.Value !== undefined)) {
 		//!steal-remove-start
 
-		// If value is an object or array, give a warning 
-		if (definition.value !== null && typeof definition.value === 'object') {
-			dev.warn("can-define: The value for options is set to an object. This will be shared by all instances of the DefineMap. Use a function that returns the object instead.");
+		// If value is an object, array, or constructor give a warning
+		if (definition.value !== null && (typeof definition.value === 'object' || canReflect.isConstructorLike(definition.value || {}))) {
```

Comment:
The two cases (plain object & constructor) should have separate warnings.  It's likely that if a constructor is passed, the user meant to use `Value` instead of `value`, and that would be better reflected in a warning like https://github.com/canjs/can-define/blob/21787bb1021c464cdff4b8d40c40c9bfe045526a/can-define.js#L187